components/gh_events/GitHubEvents.tsx

```diff
@@ -27,17 +26,29 @@ export default function GitHubEvents({ minimal }: { minimal?: boolean }) {
   const [events, setEvents] = useState<IGitHubEvent[]>();
 
   useEffect(() => {
-    fetch(
-      `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/events?per_page=100&page=${page}`,
-    )
-      .then((res) => res.json())
-      .then((data) =>
-        setEvents(
-          combineSimilarPushEvents(
-            data.filter(exludeBotEvents).filter(excludeBlacklistedEvents),
-          ).slice(0, 5),
-        ),
-      );
+    const fetchEvents = async () => {
+      try {
+        const response = await fetch(
+          `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/events?per_page=100&page=${page}`,
+        );
+        if (!response.ok) {
+          throw new Error("Failed to fetch data");
+        }
+        const data = await response.json();
+        const filteredData = data
+          .filter(exludeBotEvents)
+          .filter(excludeBlacklistedEvents);
+        const combinedEvents = combineSimilarPushEvents(filteredData).slice(
+          0,
+          5,
+        );
+        setEvents(combinedEvents);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+
+    fetchEvents();
```

Comment:
Why were these modified?

Reply:
Ohh!, I forgot to mention that .I noticed that there is use old method to fetch api .
So I use new and I thought for this small changes there is no need to raise issue so i changed it . Should I revert it or it's okk?

Reply:
Revert and keep it as a separate PR 😄 

- Would be easier to review and test. 
- If something is wrong, it'd be easier to pin-point from where related issue/bug came from.
- Also gives you more points in PRs opened/merged in the leaderboard! 😜

Reply:
Raise issue for it too, so that one could know what problem you are trying to solve by looking at the issue

Reply:
Okay I will raise a separate issue and separate pr for this. 😅Btw,  Thanks nice idea 💡